cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -1165,6 +1240,32 @@ func (s *Supervisor) onMessage(ctx context.Context, msg *types.MessageData) {
 		}
 	}
 
```

Comment:
Nit: the onMessage func is pretty long and hard too read. Would be nice to split into smaller funcs to improve readability.